### PR TITLE
Added Ember Fire in frontend to provide authentication

### LINKED ISSUE #758: Using Ember Firebase Data Adapter in Frontend.
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

Integration of Firebase Adapter in the frontend in Refractor Frontend Part.  Library : https://github.com/firebase/emberfire

__Yes__ Sending PR for the same.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #758 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Provide authentication using ember fire
-

### CODE REVIEW

**Comment by djmgit** [comment]
@yashLadha travis is failing , please have a look.

File: frontend/config/environment.js [hunk 1]
```diff
@@ -5,7 +5,23 @@ module.exports = function(environment) {
     modulePrefix : 'badgeyay',
     environment,
     rootURL      : '/',
-    locationType : 'hash',
+    locationType : 'auto',
+    firebase: {
+      apiKey: "",
```
[the remark this answers is not in the document]
Comment (harshit98) [reply frontend/config/environment.js]: You can comment out the lines from 9 to 16. Once you setup the auth features with firebase, you can uncomment those lines having each key it's proper value.

**Comment by harshit98** [comment]
@yashLadha Please go through my review I've left for you.

**Comment by djmgit** [comment]
@yashLadha also please update your commit message, you must mention which issue you are solving. Please follow the commit structure of FOSSASIA.

**Comment by djmgit** [comment]
@yashLadha its great that you are working on firebase auth, but I can see that @vaibhavsingh97 had shown interest to work on this issue in #707. @yashLadha I would request you that before sending any PR please make sure whether anybody else is working on it or not (especially if mentioned in the issue itself). Issues might not always be assigned, but if someone has shown interest in working, please try to at least make a comment like 'is anyone working on this issue?' or something like that.
Please follow best practices

**Comment by yashLadha** [comment]
@djmgit I have done the signup part because he has also shown interest in Integrating with python SDK and currently the library doesn't have support for signup with the new firebase SDK. SO he still can work on the same. After the discussion in the meeting on 14, he hasn't updated the issue so I thought to send the PR for the same. Still, he is work on the Integration with the Python SDK and creates a signup functionality. I will take care that in future to have no conflicts.

**Comment by yashLadha** [comment]
Done changes as requested @harshit98 @djmgit

**Comment by yashLadha** [comment]
Ping @djmgit Please review and merge the same.

**Comment by yashLadha** [comment]
Please merge @djmgit so that this PR can be merged https://github.com/fossasia/badgeyay/pull/768

**Comment by yashLadha** [comment]
Done @djmgit

**Comment by yashLadha** [comment]
@djmgit please merge it so that https://github.com/fossasia/badgeyay/pull/768 can be merged. and the basic sign in functionality gets set up.

**Comment by yashLadha** [comment]
Ping @harshit98 can you merge the same because other PRs are stalled because of this PR and this https://github.com/fossasia/badgeyay/pull/768 Thanks

**Pull request merged**